python/cudnn/sdpa/fwd/api_dsl.py

```
@@ -1492,29 +1476,35 @@ def _execute_mxfp8(
-            else self._dummy("seq_kv", device, lambda: torch.zeros(b, dtype=torch.int32, device=device))
-        )
-        o_desc_dummy = self._dummy("o_desc", device, lambda: torch.zeros(1, dtype=torch.int64, device=device))
+        # Tensor prep can LAUNCH work (_to_bshd's gather copy, _reshape_sf's
+        # .contiguous(), the dummies' first-use zero-fill) — launch stream
+        # (Rule 5), like the amax reset below.
+        with _torch_stream_context(current_stream, device):
+            Q = self._to_bshd(q_tensor)
+            K = self._to_bshd(k_tensor)
+            V = self._to_bshd(v_tensor)
+            O_view, o_needs_copy_back, O_scratch = self._to_bshd_writable(o_tensor)
+            O = O_scratch if o_needs_copy_back else O_view
```

Ruff (0.16.1)

[error] Ambiguous variable name: `O`

(E741)